src/api/include/migraphx/migraphx.hpp

```diff
@@ -445,9 +445,6 @@ struct shape : MIGRAPHX_CONST_HANDLE_BASE(shape)
 {
     shape() {}
 
-    MIGRAPHX_DEPRECATED("Contructor without lifetime annotation is deprecated.")
-    shape(const migraphx_shape* p) { this->set_handle(p, borrow{}); }
```

Comment:
Why is this constructor removed? It is deprecated but I would like to wait until rocm 6.0 to remove it.

Reply:
While setting function pointer for the compute_method, `auto_convert_param` matches with this constructor and `-Werror` converts warning into error failing the build. 

../../../src/api/include/migraphx/migraphx.hpp:389:10: error: ‘migraphx::api::shape::shape(const migraphx_shape*)’ is deprecated: Contructor without lifetime annotation is deprecated. [-Werror=deprecated-declarations]
     auto auto_convert_param(rank<1>, T x) -> decltype(as_handle<T>{x})
          ^~~~~~~~~~~~~~~~~~
../../../src/api/include/migraphx/migraphx.hpp:451:5: note: declared here
     shape(const migraphx_shape* p) { this->set_handle(p, borrow{}); }

Reply:
Added back that constructor, the thing that was causing it was c++ shape is a `const` handle base type and `C` API was passing non-const type. For auto_convert_param to match and invoke a constructor with lifetime at `rank<2>`, I had to pass `const_migraphx_shape` from C API.